app/src/main/java/tttTDD/Java/Game.java

```diff
@@ -67,11 +67,21 @@ public boolean performMove(int playerInput) {
         return true;
     }
 
-    private boolean gameHasWinner() {
-        if (board.hasWinningSet(currentPlayer.getMarker())) {
+    private boolean gameHasFinished() {
+        if (isThereAWinner() && !isThereATie()) {
             gameWinner = String.format("Player %s has won", currentPlayer.getMarker());
+        } else {
+            gameWinner ="";
         }
 
+        return isThereAWinner() || isThereATie();
+    }
+
+    private boolean isThereATie() {
```

Comment:
You have a business/domain concept in the project being asked for and a specific method to relate it to. This is a good addition. Even though I said for simple conditionals I wouldn't create wrapping methods for them in an earlier statement, you found the exact reason that they would be done and is a good example of expressing the rules of task directly in code. That always trumps individual styling preferences.

Reply:
lol, well that sounds like you can add your name to that list also. :)